File: recipes_api/models/recipe-model.js
"use strict";


module.exports = class Recipe {

    constructor(recipe) {
        if (recipe) {
            this.id = recipe.id || "";
            this.parent = recipe.parent || null;
            this.name = recipe.name || "";
            this.description = recipe.description || "";
            this.stars = recipe.stars || 0;
        } else {
            this.id = "";
            this.parent = "";
            this.name = "";
            this.description = "";
            this.stars = 0;
        }
    }

    parseFromDB(row){
         let recipe = new Recipe();
         if(row){
            recipe.id = row['recipe_id'];
            recipe.parent = row['recipe_parent'] || null;
            recipe.name = row['recipe_name'] || '';
            recipe.description = row['recipe_description'] || '';
            recipe.stars = row['recipe_stars'];
         }
         return recipe;
    }
}

File: recipes/recipe-model.py
class Recipe:
    def __init__(self, recipe=None):
        if recipe:
            self.id = recipe.get('id', "")
            self.parent = recipe.get('parent', None)
            self.name = recipe.get('name', "")
            self.description = recipe.get('description', "")
            self.stars = recipe.get('stars', 0)
        else:
            self.id = ""
            self.parent = None
            self.name = ""
            self.description = ""
            self.stars = 0

    @classmethod
    def parse_from_db(cls, row):
        recipe = cls()
        if row:
            recipe.id = row.get('recipe_id', "")
            recipe.parent = row.get('recipe_parent', None)
            recipe.name = row.get('recipe_name', "")
            recipe.description = row.get('recipe_description', "")
            recipe.stars = row.get('recipe_stars', 0)
        return recipe